test/contrib/aws/instrumentation_test.rb

```diff
@@ -0,0 +1,68 @@
+require 'helper'
+require 'aws-sdk'
+require 'ddtrace'
+require 'ddtrace/contrib/aws/patcher'
+require 'ddtrace/ext/http'
+
+module Datadog
+  module Contrib
+    module Aws
+      class InstrumentationTest < Minitest::Test
+        def setup
+          # initializes the tracer and a stub client
+          @tracer = enable_tracer
+          @client = ::Aws::S3::Client.new(stub_responses: true)
+        end
+
+        def test_pin_defaults
+          assert_equal('aws', pin.service)
+          assert_equal('web', pin.app_type)
+          assert_nil(pin.app)
+          assert_nil(pin.name)
+          assert_nil(pin.config)
+        end
+
+        def test_list_buckets
+          @client.list_buckets
+          spans = @tracer.writer.spans
+          assert_equal(1, spans.length)
```

Comment:
@p-lambert always grab the `spans` list and check how many spans are there. If you search only what you need, you will miss:
* double instrumentation (2 spans instead of one)
* conflicts in instrumentation (one integration triggers wrongly another one)
* wrong parenting if you're expecting 2+ spans